container-aws-go/Pulumi.yaml

```diff
@@ -0,0 +1,18 @@
+name: ${PROJECT}
+description: ${DESCRIPTION}
+runtime: go
+template:
+  description: A containerized service on AWS Fargate
+  config:
```

Comment:
I had expected (at least) two other config variables:
1) A `imageName`, in case you already have an image to deploy. That would let you have a fully functional deployment without needing code edits - e.g., just configure it to have `nginx`.

2) A `replicas` count, so you can configure and scale up/down the service.

Maybe others as I look through but those leapt out to me.

Reply:
> A imageName, in case you already have an image to deploy. 

The template currently does a build and push from a local folder, so that users can configure this by changing what's there instead of hard coding an image name.  My sense is this is much more "real", and makes it much easier to say "just put your code in ./app and re-run pulumi up" - which we can't say if we have this set via a fixed image name.

> That would let you have a fully functional deployment without needing code edits - e.g., just configure it to have nginx.

Note that you do have a fully functional `nginx` deployment out of the box, because the default contents of `./app` is a `Dockerfile` with `FROM nginx`.  So to a meaningful degree - you get the best of both worlds - something that works out of the box, and is easy to "make yours" by just dropping your app code in a folder.

We could in theory support "both", but it would make the template a lot more confusing.  On balance, I prefer the approach here currently.

Though I do think it makes the stronger case for the need for READMEs.

> A replicas count, so you can configure and scale up/down the service.

A sort of meta-question is what value exactly you perceive there being in having lots of config for these templates.  I started off with more config, but realized there is fairly limited value, because the only way to edit config after the initial deployment is to make an edit to the repo, and it's just as easy to edit the code as to edit the config.  The real role of config is to make it easier to have things that will differ across different stacks deployed from the same project.  

The other factor is that it's easy to go overboard with config and just make things very complicated for little value.  My sense after trying a few versions was that it's better to be simple out of the box.